index/scorch/persister.go

```diff
@@ -583,6 +583,7 @@ func (s *Scorch) persistSnapshotDirect(snapshot *IndexSnapshot) (err error) {
 	for _, filename := range filenames {
 		delete(s.ineligibleForRemoval, filename)
 	}
+	s.stats.CurFilesIneligibleForRemoval = uint64(len(s.ineligibleForRemoval))
```

Comment:
maybe instead of carefully maintaining CurFilesIneligibleForRemoval every time s.ineligibleForRemoval is updated...

instead during the stats call read the latest len()?

Reply:
yep. make sense!